source/extensions/transport_sockets/proxy_protocol/proxy_protocol.cc

```diff
@@ -165,6 +168,69 @@ void UpstreamProxyProtocolSocketFactory::hashKey(
   }
 }
 
+std::unordered_map<uint8_t, std::vector<unsigned char>> UpstreamProxyProtocolSocket::buildCustomTLVs() {
```

Comment:
Note, I previously tried populating this data in the UpstreamProxyProtocolSocket constructor, but the issue I was seeing with that approach is the transport socket callbacks haven't been populated until after construction, and therefore we can't read from the filter metadata namespace at that point in the function chain.